staging/src/k8s.io/apiserver/pkg/storage/cacher/store_btree.go

```diff
@@ -385,3 +393,79 @@ func (i *indexer) delete(key, value string, index map[string]map[string]*storeEl
 		delete(index, value)
 	}
 }
+
+// storeSnapshotter caches snapshots of store created by cloning the store.
+type storeSnapshotter struct {
+	mux       sync.RWMutex
+	snapshots *btree.BTreeG[rvSnapshot]
```

Comment:
> Yes, this comes from PoC where I knew I need a ordered map to lookup DescendLessOrEqual element. Used btree because it was already available. Think this could be further optimized (cyclic buffer for rev + map[rev]snapshot), but I don't thinks we should block on it.

I'm fine with not blocking on further improvements, but please add a comment explaining that we effectively only need ordering and we can potentially simplify that in the future.